builtin/game/chatcommands.lua

```diff
@@ -451,59 +457,123 @@ core.register_chatcommand("set", {
 	end,
 })
 
-local function emergeblocks_callback(pos, action, num_calls_remaining, ctx)
-	if ctx.total_blocks == 0 then
-		ctx.total_blocks   = num_calls_remaining + 1
-		ctx.current_blocks = 0
+local function emerge_callback(pos, action, num_calls_remaining, ctx)
+	if ctx.total_count == 0 then
+		ctx.total_count   = num_calls_remaining + 1
+		ctx.current_count = 0
 	end
-	ctx.current_blocks = ctx.current_blocks + 1
-
-	if ctx.current_blocks == ctx.total_blocks then
+	ctx.current_count = ctx.current_count + 1
+	if os.clock() < ctx.cur_time then
+		-- os.clock() loops after 2^32 microseconds.
```

Comment:
`os.clock()` is a wrapper around the C `clock()` function, which returns a 32-bit value (in any case on 32-bit platforms, it does). On Posix, it loops every 2^32 *micro*seconds (but the first time after 2^31 seconds), on windows it loops every 2^32 *milli*seconds. I don't know about other platforms or 64-bit platforms.

Apparently, the behavior of `os.clock()` is not consistent across platforms, in that it may return CPU-time consumed (Posix), or a wall-clock time since program start (on Windows, reportedly). So it's probably best to replace it altogether.

I don't prefer `os.time()` here, as it has second-resolution only. To make the behavior consistent on the different platforms, I opted for `minetest.get_us_time()`. On Posix, this changes the semantics, but on Windows it shouldn't.

`minetest.get_us_time()` is 32-bit as well, so it loops after 72 minutes. So that part of the code stays. As it's not critical here, the overflow handing code is basic and simple, but it may not always be exact.